File: Orientacion_a_objetos_1/ProyectoGuia1y2/src/modelo/Persona.java
package modelo;

public class Persona {
	
	//ATRIBUTOS
	private String nombre;
	private String apellido;
	private int dni;
	private Contacto contacto;
	
	//CONSTRUCTOR
	public Persona(String nombre, String apellido, int dni, Contacto contacto) {
		super();
		this.nombre = nombre;
		this.apellido = apellido;
		this.dni = dni;
		this.contacto = contacto;
	}

	//GETTER Y SETTER
	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getApellido() {
		return apellido;
	}

	public void setApellido(String apellido) {
		this.apellido = apellido;
	}

	public int getDni() {
		return dni;
	}

	public void setDni(int dni) {
		this.dni = dni;
	}

	public Contacto getContacto() {
		return contacto;
	}

	public void setContacto(Contacto contacto) {
		this.contacto = contacto;
	}

	//toString y equals
	@Override
	public String toString() {
		return "Persona [nombre=" + nombre + ", apellido=" + apellido + ", dni=" + dni + ", contacto=" + contacto + "]";
	}
	
	public boolean equals(Persona persona) {
		return dni == persona.getDni();
	}
	
	//OTROS METODOS
}

File: Algoritmo_y_estructura_de_datos/Final/Final1/persona.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "nodo.h"
#include "persona.h"

struct Persona{
    char nombre[15];
    int dni;
};

PersonaPtr crearPersona(char n[15], int d){
    PersonaPtr p = (PersonaPtr) malloc(sizeof(struct Persona));
    strcpy(p->nombre, n);
    p->dni = d;

    return p;
};

//Getter
char* getNombre(PersonaPtr p){
    return p->nombre;
};

int getDni(PersonaPtr p){
    return p->dni;
};

//Setter
void setNombre(PersonaPtr p, char n[15]){
    strcpy(p->nombre, n);
};

void setDni(PersonaPtr p, int nuevoDni){
    p->dni = nuevoDni;
};

void wrapperPersona(void * dato){
    PersonaPtr p = (PersonaPtr) dato;

    mostrarPersona(p);
};

void mostrarPersona(PersonaPtr p){
    printf("\n Persona: Nombre:  %s - DNI: %d \n", p->nombre, p->dni);
};


